### PR TITLE
Add exclude_filetypes

### LINKED ISSUE #97: Exclude a specific filetypes
Thank you for the cool plugin! I use this plugin in LISP code.
In LISP code, quote `'` is not a paired character. Thus, I write the following code in my `init.lua`.

```lua
pairs.remove_rule("'")↲
pairs.add_rule(
  rule("'", "'")↲
    :with_pair(cond.not_before_regex_check("%w"))↲
    :with_pair(function(_)↲
        return vim.bo.filetype ~= "lisp" and vim.bo.filetype ~= "scheme"↲
    end)
)
```

I propose a new condition like `exclude_filetypes({"lisp"})`.
Would you like to add this condition to `lua/nvim-autopairs/conds.lua`?

### PR DESCRIPTION
**Pull request by tani**
Closes #97

### CODE REVIEW

File: lua/nvim-autopairs/conds.lua [hunk 1]
```diff
@@ -199,4 +198,16 @@ cond.is_end_line = function()
...
 
+cond.exclude_filetypes = function(filetypes)
+    return function(opts)
+        log.debug('exclude_filetypes')
+        for _, filetype in pairs(filetypes) do
+            if vim.bo.filetype == filetype then
+                return false
+            end
+        end
+        return true
```
Comment (windwp) [inline lua/nvim-autopairs/conds.lua:209]: you don't need return true here i already say a condition have 3 state to return.

it need to return nil when you are not sure it is correct on the condition

you can check another "not_" condition it doesn't need to return true
Can you change the name to "not_filetypes" to make it similar to other condition

another code is looking good. thank you

**Review by windwp — CHANGES_REQUESTED** [review]

File: lua/nvim-autopairs/conds.lua (hunk 1 above)
Comment (tani) [reply lua/nvim-autopairs/conds.lua]: done

**Comment by tami5** [comment]
@tani can you share your configuration. My setting doesn't seem to be working

```lua

return function()
  local npairs = require "nvim-autopairs"
  local Rule = require "nvim-autopairs.rule"
  local cond = require "nvim-autopairs.conds"
  npairs.setup {
    blank_line_filetype = nil,
    pairs_map = {
      -- ["'"] = "'",
      ['"'] = '"',
      ["("] = ")",
      ["["] = "]",
      ["{"] = "}",
      ["`"] = "`",
    },
    disable_filetype = { "TelescopePrompt" },
    html_break_line_filetype = { "html", "vue" },
    ignore_next_char = "[%w%.%+%-%=%/%,\"'{}%[%]]",
  }
  npairs.add_rule(Rule("'", "'"):with_pair(cond.not_filetypes { "clojure", "clojurescript", "fennel" }))
  npairs.add_rule(Rule("|", "|", "rust"))j
end
```

**Comment by ignamartinoli** [comment]
@kkharji +1 I'm also trying to disable that pair for **Lisp** files

**Comment by gnsfujiwara** [comment]
> @kkharji +1 I'm also trying to disable that pair for **Lisp** files

I do this in my configuration:

```lua
-- Don't close single quotes in lisps
autopairs.add_rules({
  rule("'", '', {
    'lisp',
    'clojure',
    'scheme',
    'fennel',
  }),
})
```

**Comment by ignamartinoli** [comment]
Thanks @gnsfujiwara, I was trying to accomplish it. When I have free time I'm going to add some of this rules as default for some languages with a PR

**Comment by tani** [comment]
Dear @ignamartinoli ,  @gnsfujiwara , and @kkharji ,

I'm sorry for the late response. You're missing to remove an existing rule.
 
```lua
local autopairs = require 'nvim-autopairs'
local rule = require 'nvim-autopairs.rule'
local conds = require 'nvim-autopairs.conds'
autopairs.setup()
autopairs.remove_rule("'")
autopairs.add_rule(rule("'", "'"):with_pair(conds.not_filetypes({'clojure'})))

```

**Comment by windwp** [comment]
https://github.com/windwp/nvim-autopairs#plugin-integration
you can remove rule and add new rule or get_rule then modify it

**Pull request merged**